mutation-testing/check-packages-and-shards/action.yml

```diff
@@ -0,0 +1,123 @@
+    value: ${{ steps.check_packages_and_shards.outputs.big_packages_with_shards }}
+  run_small_packages:
+    value: ${{ steps.check_packages_and_shards.outputs.run_small_packages }}
+  small_packages_with_shards:
+    value: ${{ steps.check_packages_and_shards.outputs.small_packages_with_shards }}
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Checkout stacks-core repo
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Install cargo-mutants
+      shell: bash
+      run: cargo install --version 23.12.2 cargo-mutants
+
+    - name: Relative diff
+      shell: bash
+      run: |
+        git diff origin/${{ github.base_ref }}.. > git.diff
+
+    - name: Update git diff
+      shell: bash
+      run: |
+        input_file="git.diff"
+        temp_file="temp_diff_file.diff"
+
+        # Check if the commands exist on the host
+        for cmd in tac awk sed; do
+          command -v "${cmd}" > /dev/null 2>&1 || echo "Missing command: ${cmd}"
+        done
+
+        # Reverse the file, remove 4 lines after '+++ /dev/null', then reverse it back (editors can't go backwards - to remove lines above)
+        tac "$input_file" > "$temp_file" && mv "$temp_file" "$input_file"
+        sed '/+++ \/dev\/null/{n;N;N;N;d;}' "$input_file" > "$temp_file" && mv "$temp_file" "$input_file"
+        tac "$input_file" > "$temp_file" && mv "$temp_file" "$input_file"
+
+        # Remove the lines between '+++ /dev/null' (included) and 'diff --git a/'
+        awk '
+          BEGIN { in_block=0 }
+          /\+\+\+ \/dev\/null/ { in_block=1; next }
+          in_block && /diff --git a\// { in_block=0; print; next }
+          !in_block
+        ' "$input_file" > "$temp_file" && mv "$temp_file" "$input_file"
+
+    - name: Split diffs
+      shell: bash
+      run: |
+        cargo mutants --in-diff git.diff --list > all_mutants.txt
+        mkdir -p mutants_by_packages
+
+        # Check that the file exists before performing actions on it
+        if [ ! -s all_mutants.txt ]; then
+          echo "The file containing mutants is missing or empty!"
+          exit 1
+        fi
+
+        # Split the differences from git into 2 parts, big packages ('stacks-node' and 'stackslib') and small packages (all others) and put them into separate files
+        while IFS= read -r line; do
+          package=$(echo "$line" | cut -d'/' -f1)
+          if [[ $package == "testnet" || $package == "stackslib" ]]; then
+            echo "$line" >> "mutants_by_packages/big_packages.txt"
+          else
+            echo "$line" >> "mutants_by_packages/small_packages.txt"
+          fi
+        done < all_mutants.txt
+
+    - name: Check packages and shards
+      id: check_packages_and_shards
+      shell: bash
+      run: |
+        number_of_big_mutants=0
+        number_of_small_mutants=0
+
+        # If big_packages file exists, count how many mutants there are
+        if [[ -s mutants_by_packages/big_packages.txt ]]; then
```

Comment:
i don't see an else here - maybe move this to a negation condtional that exits. 
ex:
```bash
if [[ ! -s ... ]; then
  echo "error"
  exit 1
fi
number_of_big_mutants=$(cat...
```

Reply:
In this situation we don't need an else, since if there aren't mutations on big packages, we can still have them on the small ones, so we don't need to exit early.